test/suites/utilization/fips_test.go

```diff
@@ -0,0 +1,239 @@
+				readyCondition := nodeClass.StatusConditions().Get(opstatus.ConditionReady)
+				g.Expect(readyCondition.IsFalse()).To(BeTrue())
+			}).Should(Succeed())
+		})
+	})
+
+	Context("FIPS Provisioning", func() {
+		BeforeEach(func() {
+			if env.InClusterController {
+				Skip("FIPS tests require SIG access - skipping in self-hosted mode")
+			}
+		})
+
+		It("should provision FIPS-enabled Ubuntu nodes", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.UbuntuImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr(v1beta1.FIPSModeFIPS)
+
+			pod := test.Pod()
+			env.ExpectCreated(nodeClass, nodePool, pod)
+			env.EventuallyExpectHealthy(pod)
+			node := env.ExpectCreatedNodeCount("==", 1)[0]
+
+			imageRef := validateNodeUsesFIPS(node)
+			validateNodeClassHasExpectedImages(imageRef, true) // true = expect FIPS images
+		})
+
+		It("should provision FIPS-enabled AzureLinux nodes", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr(v1beta1.FIPSModeFIPS)
+
+			pod := test.Pod()
+			env.ExpectCreated(nodeClass, nodePool, pod)
+			env.EventuallyExpectHealthy(pod)
+			node := env.ExpectCreatedNodeCount("==", 1)[0]
+
+			imageRef := validateNodeUsesFIPS(node)
+			validateNodeClassHasExpectedImages(imageRef, true) // true = expect FIPS images
+		})
+	})
+
+	Context("FIPS Disabled", func() {
+		It("should provision nodes with FIPS explicitly disabled", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr(v1beta1.FIPSModeDisabled)
+
+			pod := test.Pod()
+			env.ExpectCreated(nodeClass, nodePool, pod)
+			env.EventuallyExpectHealthy(pod)
+			node := env.ExpectCreatedNodeCount("==", 1)[0]
+
+			imageRef := validateNodeDoesNotUseFIPS(node)
+			validateNodeClassHasExpectedImages(imageRef, false) // false = expect non-FIPS images
+		})
+	})
+})
+
+// validateNodeUsesFIPS checks that the node is using a FIPS-compliant image and returns the image reference
+func validateNodeUsesFIPS(node *v1.Node) string {
+	GinkgoHelper()
+
+	// Extract VM name from the node's providerID
+	vmName, err := nodeclaim.GetVMName(node.Spec.ProviderID)
+	Expect(err).ToNot(HaveOccurred(), "Should be able to extract VM name from providerID")
+
+	// Get the VM using the extracted name
+	vm := env.GetVMByName(vmName)
+	Expect(vm.Properties).ToNot(BeNil())
+	Expect(vm.Properties.StorageProfile).ToNot(BeNil())
+	Expect(vm.Properties.StorageProfile.ImageReference).ToNot(BeNil())
+
+	// Extract the image reference
+	imageReference := utils.ImageReferenceToString(vm.Properties.StorageProfile.ImageReference)
+	Expect(imageReference).ToNot(BeEmpty(), "VM should have an image reference")
+
+	// FIPS images require SIG (Shared Image Gallery) access
+	// SIG images have ID field populated, while CIG (Community Gallery)
+	// images use CommunityGalleryImageID
+	isSIGImage := vm.Properties.StorageProfile.ImageReference.ID != nil &&
```

Comment:
minor: You can use `lo.FromPtr(vm.Properties.StorageProfile.ImageReference.ID) != ""` here to make this shorter.